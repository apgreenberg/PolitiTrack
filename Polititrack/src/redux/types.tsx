import { UserInfo } from '../model/database/userInterfaces';
import { ThunkAction, ThunkDispatch } from 'redux-thunk';

import { store } from './store';

//Action types
export const LOGIN = 'LOGIN';
export const LOGOUT = 'LOGOUT';

export interface LoginAction {
  type: typeof LOGIN;
  payload: UserInfo;
}

export interface LogoutAction {
  type: typeof LOGOUT;
}

export type Actions = LoginAction | LogoutAction;

//State interface typing
export interface StateWrapper {
  loggedIn: boolean;
  currentUser: UserInfo;
}

//Dispatch types
export type ThunkActionType<ReturnType = void> = ThunkAction<
  ReturnType,
  StateWrapper,
  unknown,
  Actions
>;

export type ThunkDispatchType = ThunkDispatch<StateWrapper, undefined, Actions>;

export type AppDispatch = typeof store.dispatch;
