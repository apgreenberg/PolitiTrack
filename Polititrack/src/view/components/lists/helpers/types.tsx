import { OnboardContent } from '../../../../model/database/userInterfaces';
import { APISingleResponseNode } from '../APISingleResponseListItem';
import { APIMultiResponseNode } from '../APIMultiResponseList';

export type SwipeRowData =
  | OnboardContent
  | APISingleResponseNode
  | APIMultiResponseNode;

export interface OnPressActions {
  clickable: boolean;
  onPress: (data: SwipeRowData, displayText: string) => void;
  addable: boolean;
  addOnPress: (data: SwipeRowData, displayText: string) => void;
  deletable: boolean;
  deleteOnPress: (data: SwipeRowData, displayText: string) => void;
}
