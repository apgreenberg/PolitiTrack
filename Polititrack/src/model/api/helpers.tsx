export enum Status {
  Loading,
  Error,
  Success,
}
