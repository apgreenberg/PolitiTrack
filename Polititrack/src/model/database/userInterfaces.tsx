export interface AllUserData {
  users: UserInfo[];
}
export interface UserInfo {
  id: number;
  token: string;
  email: string;
  password: string;
  name: string;
  data: UserContent;
  settings: UserSettings;
}
export interface UserContent {
  jurisdictions: OnboardContent[];
  bills: OnboardContent[];
  billTypes: OnboardContent[];
  legislators: OnboardContent[];
}
export interface UserSettings {
  updateJurisdictions: boolean;
  updateBills: boolean;
}
export interface UserUpdates {
  jurisdictions: Date;
  bills: Date;
}
export interface OnboardContent {
  id: string;
  name: string;
  lastUpdated: number;
}
export const blankUserInfo: UserInfo = {
  id: 0,
  token: '',
  email: '',
  password: '',
  name: '',
  data: { jurisdictions: [], bills: [], billTypes: [], legislators: [] },
  settings: { updateJurisdictions: false, updateBills: false },
};
