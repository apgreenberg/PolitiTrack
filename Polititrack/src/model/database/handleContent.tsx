import { updateExternalDatabase } from './userHelpers';
import { OnboardContent, UserContent, UserInfo } from './userInterfaces';

/*
In the future, this will be generalized to work for bills and legislators as well
*/
/**
 * Adds the requested content to the user
 * @param userInfo - the user to add the content to
 *  * @param type - bills, billTypes, legislatures, or legislators
 * @param toAdd - the content to add
 */
export const addContentToDatabase = async (
  userInfo: UserInfo,
  type: keyof UserContent,
  toAdd: OnboardContent
): Promise<void> => {
  if (!alreadyContains(userInfo, type, toAdd)) {
    userInfo.data[type].push(toAdd);
    alert('Successfully added ' + toAdd.name);
  } else {
    alert('You have already added ' + toAdd.name);
  }
  await updateExternalDatabase();
};

/**
 * Removes the requested content from the user
 * @param userInfo  - the user to remove the content from
 * @param type - bills, billTypes, legislatures, or legislators
 * @param toRemoveId - the id of the content in the users list of content
 */
export const removeContentFromDatabase = async (
  userInfo: UserInfo,
  type: keyof UserContent,
  toRemoveId: string
): Promise<void> => {
  for (let i = 0; i < userInfo.data[type].length; i++) {
    if (userInfo.data[type][i].id === toRemoveId) {
      userInfo.data[type].splice(i, 1);
      break;
    }
  }
  await updateExternalDatabase();
};

/**
 * Helper function - checks if a certain content element already exists in the user's data
 * @param userInfo - the current user's info
 * @param type - the type of content to check
 * @param toAdd - the item the user is attempting to add
 * @returns - true if toAdd is already in the user's data, false otherwise
 */
export const alreadyContains = (
  userInfo: UserInfo,
  type: keyof UserContent,
  toAdd: OnboardContent
): boolean => {
  for (let content of userInfo.data[type]) {
    if (content.id === toAdd.id) {
      return true;
    }
  }
  return false;
};

/**
 * Updates the "lastUpdated" field in the content item
 * @param content - the content item to update
 */
export const updateLastUpdated = async (content: OnboardContent) => {
  content.lastUpdated = new Date().getTime();
  await updateExternalDatabase();
};
