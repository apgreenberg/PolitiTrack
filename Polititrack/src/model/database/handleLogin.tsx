import AsyncStorage from '@react-native-async-storage/async-storage';

import InternalDatabase from './userData';
import { updateExternalDatabase } from './userHelpers';
import { UserInfo } from './userInterfaces';

/**
 * Attempts to find a user in the internal database
 * @param email - the email of the user to find
 * @param password - the password of the user to find
 * @returns - the user if they are found, null otherwise
 */
export const findUser = (email: string, password: string): UserInfo | null => {
  const allUsers = InternalDatabase.users;
  for (let user of allUsers) {
    if (user.email === email && user.password === password) {
      return user;
    }
  }
  return null;
};

/**
 * Attempts to sign in the current user by updating the
 * 'currentUserInfo' record in AsyncStorage to the user's info
 * @param foundUserInfo - the info of the user to sign in
 */
export const attemptSignIn = async (foundUserInfo: UserInfo): Promise<void> => {
  try {
    await AsyncStorage.setItem(
      'currentUserInfo',
      JSON.stringify(foundUserInfo)
    );
  } catch (e) {
    console.log(e);
  }
};

/**
 * Attempts to sign out the current user by updating the
 * 'currentUserInfo' record in AsyncStorage to null
 */
export const attemptSignOut = async (): Promise<void> => {
  try {
    await AsyncStorage.removeItem('currentUserInfo');
  } catch (e) {
    console.log(e);
  }
};

/**
 * Attempts to delete a user's account
 * @param id - the id of the user's account to delete
 */
export const attemptDeleteAccount = async (id: number): Promise<void> => {
  for (let i: number = 0; i < InternalDatabase.users.length; i++) {
    if (InternalDatabase.users[i].id === id) {
      InternalDatabase.users.splice(i, 1);
    }
  }
  await updateExternalDatabase();
};

/*
Pulls the user info of the last signed in user from async storage, so user remains signed in
when reopening the app
*/
export const pullCurrentUserInfo = async (): Promise<UserInfo | null> => {
  let userInfo: UserInfo | null = null;
  let userInfoString: string | null = null;
  try {
    userInfoString = await AsyncStorage.getItem('currentUserInfo');
  } catch (e) {
    console.log(e);
  }
  if (userInfoString !== null) {
    userInfo = JSON.parse(userInfoString);
  }
  return userInfo;
};
