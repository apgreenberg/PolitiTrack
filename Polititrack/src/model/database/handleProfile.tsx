import { updateExternalDatabase } from './userHelpers';
import { UserInfo } from './userInterfaces';

export const updateProfile = (
  userInfo: UserInfo,
  type: keyof UserInfo,
  newValue: string
): void => {
  if (type === 'name' || type === 'email' || type === 'password') {
    userInfo[type] = newValue;
  }
  updateExternalDatabase();
};
