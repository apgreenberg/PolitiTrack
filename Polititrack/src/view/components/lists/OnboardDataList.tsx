import React from 'react';
import { View } from 'react-native';

import { OnboardContent } from '../../../model/database/userInterfaces';
import CustomSwipeRow from './helpers/CustomSwipeRow';

import { OnPressActions } from './helpers/types';

export interface OnboardDataListProps {
  data: OnboardContent[];
  onPressActions: OnPressActions;
}

/**
 * Displays a list of the precomputed (onboard) data
 * @param props - ListProps:
 *                -data: the data to display in the list
 *                -onPressActions: handles the clickability, addability, and deletability of each item in the list
 * @returns - a functional component representing the list
 */
const OnboardDataList: React.FC<OnboardDataListProps> = (props) => {
  return (
    <View style={{ marginBottom: 180 }}>
      {props.data.map((dataItem: OnboardContent, i: number) => (
        <CustomSwipeRow
          key={i}
          onPressActions={props.onPressActions}
          data={dataItem}
          displayText={dataItem.name}
        />
      ))}
    </View>
  );
};

export default OnboardDataList;
