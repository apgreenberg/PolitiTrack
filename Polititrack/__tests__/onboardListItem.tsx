import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { MockedProvider } from '@apollo/client/testing';

import { OnPressActions } from '../src/view/components/lists/helpers/types';
import OnboardDataList from '../src/view/components/lists/OnboardDataList';
import { OnboardContent } from '../src/model/database/userInterfaces';

/**
 * Tests the Onboard Content List
 */
const mainSwipeButtonOnPress = jest.fn();
const addSwipeButtonOnPress = jest.fn();
const deleteSwipeButtonOnPress = jest.fn();
const onPressActions: OnPressActions = {
  clickable: true,
  onPress: mainSwipeButtonOnPress,
  addable: true,
  addOnPress: addSwipeButtonOnPress,
  deletable: true,
  deleteOnPress: deleteSwipeButtonOnPress,
};
const onPressActionsNotClickable: OnPressActions = {
  clickable: false,
  onPress: mainSwipeButtonOnPress,
  addable: false,
  addOnPress: addSwipeButtonOnPress,
  deletable: false,
  deleteOnPress: deleteSwipeButtonOnPress,
};
const data: OnboardContent[] = [
  {
    id: 'id1',
    name: 'name1',
    lastUpdated: new Date().getTime(),
  },
  {
    id: 'id2',
    name: 'name2',
    lastUpdated: new Date().getTime(),
  },
  {
    id: 'id2',
    name: 'name2',
    lastUpdated: new Date().getTime(),
  },
  {
    id: 'id2',
    name: 'name2',
    lastUpdated: new Date().getTime(),
  },
];
describe('Tests the Onboard Content list', () => {
  beforeEach(() => {
    mainSwipeButtonOnPress.mockClear();
    addSwipeButtonOnPress.mockClear();
    deleteSwipeButtonOnPress.mockClear();
  });
  test('list renders correctly', async () => {
    const { toJSON } = render(
      <OnboardDataList data={data} onPressActions={onPressActions} />
    );
    expect(toJSON()).toMatchSnapshot();
  });
  test('test swipe row button clicks', async () => {
    const { getAllByTestId } = render(
      <OnboardDataList data={data} onPressActions={onPressActions} />
    );
    fireEvent(getAllByTestId('mainSwipeButton')[0], 'onPress');
    expect(mainSwipeButtonOnPress).toHaveBeenCalled();
    fireEvent(getAllByTestId('addSwipeButton')[0], 'onPress');
    expect(addSwipeButtonOnPress).toHaveBeenCalled();
    fireEvent(getAllByTestId('deleteSwipeButton')[0], 'onPress');
    expect(deleteSwipeButtonOnPress).toHaveBeenCalled();
  });
  test('test swipe row button visibility', async () => {
    const { queryByTestId } = render(
      <OnboardDataList
        data={data}
        onPressActions={onPressActionsNotClickable}
      />
    );
    expect(queryByTestId('addSwipeButton')).toBeNull();
    expect(queryByTestId('deleteSwipeButton')).toBeNull();
  });
});
