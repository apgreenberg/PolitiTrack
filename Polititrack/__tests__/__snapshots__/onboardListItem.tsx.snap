// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests the Onboard Content list list renders correctly 1`] = `
<View
  style={
    Object {
      "marginBottom": 180,
    }
  }
>
  <View
    style={Object {}}
  >
    <View
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 1,
          },
          Object {
            "height": 0,
            "width": 0,
          },
        ]
      }
    >
      <View
        leftActionActivated={false}
        leftActionState={false}
        rightActionActivated={false}
        rightActionState={false}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#9ca7b8",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingLeft": 15,
          }
        }
        swipeAnimatedValue={0}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#216ba3",
              "bottom": 0,
              "justifyContent": "center",
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "top": 0,
              "width": 75,
            }
          }
          testID="addSwipeButton"
        >
          <Text
            style={
              Object {
                "color": "#FFF",
              }
            }
          >
            Add
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "red",
              "bottom": 0,
              "justifyContent": "center",
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 75,
            }
          }
          testID="deleteSwipeButton"
        >
          <Text
            style={
              Object {
                "color": "#FFF",
              }
            }
          >
            Delete
          </Text>
        </View>
      </View>
    </View>
    <View
      manipulationModes={
        Array [
          "translateX",
        ]
      }
      onLayout={[Function]}
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Object {
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
          "zIndex": 2,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#f0f0f0",
            "borderBottomColor": "#9ca7b8",
            "borderBottomWidth": 1,
            "height": 60,
            "justifyContent": "center",
          }
        }
        testID="mainSwipeButton"
      >
        <View>
          <Text
            style={
              Object {
                "fontSize": 16,
                "textAlign": "center",
              }
            }
          >
            name1
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={Object {}}
  >
    <View
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 1,
          },
          Object {
            "height": 0,
            "width": 0,
          },
        ]
      }
    >
      <View
        leftActionActivated={false}
        leftActionState={false}
        rightActionActivated={false}
        rightActionState={false}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#9ca7b8",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingLeft": 15,
          }
        }
        swipeAnimatedValue={0}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#216ba3",
              "bottom": 0,
              "justifyContent": "center",
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "top": 0,
              "width": 75,
            }
          }
          testID="addSwipeButton"
        >
          <Text
            style={
              Object {
                "color": "#FFF",
              }
            }
          >
            Add
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "red",
              "bottom": 0,
              "justifyContent": "center",
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 75,
            }
          }
          testID="deleteSwipeButton"
        >
          <Text
            style={
              Object {
                "color": "#FFF",
              }
            }
          >
            Delete
          </Text>
        </View>
      </View>
    </View>
    <View
      manipulationModes={
        Array [
          "translateX",
        ]
      }
      onLayout={[Function]}
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Object {
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
          "zIndex": 2,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#f0f0f0",
            "borderBottomColor": "#9ca7b8",
            "borderBottomWidth": 1,
            "height": 60,
            "justifyContent": "center",
          }
        }
        testID="mainSwipeButton"
      >
        <View>
          <Text
            style={
              Object {
                "fontSize": 16,
                "textAlign": "center",
              }
            }
          >
            name2
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={Object {}}
  >
    <View
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 1,
          },
          Object {
            "height": 0,
            "width": 0,
          },
        ]
      }
    >
      <View
        leftActionActivated={false}
        leftActionState={false}
        rightActionActivated={false}
        rightActionState={false}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#9ca7b8",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingLeft": 15,
          }
        }
        swipeAnimatedValue={0}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#216ba3",
              "bottom": 0,
              "justifyContent": "center",
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "top": 0,
              "width": 75,
            }
          }
          testID="addSwipeButton"
        >
          <Text
            style={
              Object {
                "color": "#FFF",
              }
            }
          >
            Add
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "red",
              "bottom": 0,
              "justifyContent": "center",
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 75,
            }
          }
          testID="deleteSwipeButton"
        >
          <Text
            style={
              Object {
                "color": "#FFF",
              }
            }
          >
            Delete
          </Text>
        </View>
      </View>
    </View>
    <View
      manipulationModes={
        Array [
          "translateX",
        ]
      }
      onLayout={[Function]}
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Object {
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
          "zIndex": 2,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#f0f0f0",
            "borderBottomColor": "#9ca7b8",
            "borderBottomWidth": 1,
            "height": 60,
            "justifyContent": "center",
          }
        }
        testID="mainSwipeButton"
      >
        <View>
          <Text
            style={
              Object {
                "fontSize": 16,
                "textAlign": "center",
              }
            }
          >
            name2
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={Object {}}
  >
    <View
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 1,
          },
          Object {
            "height": 0,
            "width": 0,
          },
        ]
      }
    >
      <View
        leftActionActivated={false}
        leftActionState={false}
        rightActionActivated={false}
        rightActionState={false}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#9ca7b8",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingLeft": 15,
          }
        }
        swipeAnimatedValue={0}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#216ba3",
              "bottom": 0,
              "justifyContent": "center",
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "top": 0,
              "width": 75,
            }
          }
          testID="addSwipeButton"
        >
          <Text
            style={
              Object {
                "color": "#FFF",
              }
            }
          >
            Add
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "red",
              "bottom": 0,
              "justifyContent": "center",
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 75,
            }
          }
          testID="deleteSwipeButton"
        >
          <Text
            style={
              Object {
                "color": "#FFF",
              }
            }
          >
            Delete
          </Text>
        </View>
      </View>
    </View>
    <View
      manipulationModes={
        Array [
          "translateX",
        ]
      }
      onLayout={[Function]}
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Object {
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
          "zIndex": 2,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#f0f0f0",
            "borderBottomColor": "#9ca7b8",
            "borderBottomWidth": 1,
            "height": 60,
            "justifyContent": "center",
          }
        }
        testID="mainSwipeButton"
      >
        <View>
          <Text
            style={
              Object {
                "fontSize": 16,
                "textAlign": "center",
              }
            }
          >
            name2
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
